### PR TITLE
Woo: Add updates to product action-list

### PR DESCRIPTION
**Pull request by coderkevin**
This adds updates for both Products and Variations to the action list,
making it suitable for editing existing products/variations and updating
them via the API.

To test: `npm test`

### CODE REVIEW

File: client/extensions/woocommerce/state/data-layer/ui/products/test/index.js [hunk 1]
```diff
@@ -332,6 +334,127 @@ describe( 'handlers', () => {
...
+				variation: variationBlack,
+			} ) );
+		} );
+
+		it( 'should update a variation for an existing product', () => {
+			const product1 = { id: 42, name: 'Product #1', type: 'variable' };
+			const variationBlack = { id: 202, sku: 'old-sku', attributes: [ { name: 'Color', options: 'Black' } ] };
+			const variationBlackEdit = { id: 202, sku: 'new-sku' };
+
+			const rootState = {
```
Comment (belcherj) [inline client/extensions/woocommerce/state/data-layer/ui/products/test/index.js]: this is mostly the same as the one above, any way to combine.  Not a deal breaker just makes it easier to see what's important about this state tree

Comment (coderkevin) [inline client/extensions/woocommerce/state/data-layer/ui/products/test/index.js]: Yeah, I think this test file probably needs to have a separate fixture added. But I think I'd like to save that for post-release.

File: client/extensions/woocommerce/app/products/product-header.js [hunk 2]
```diff
@@ -13,21 +13,51 @@ import ActionHeader from 'woocommerce/components/action-header';
...
+}
+
+function renderSaveButton( onSave, product, isBusy, translate ) {
 	const saveExists = 'undefined' !== typeof onSave;
 	const saveDisabled = false === onSave;
 
-	const saveButton = saveExists &&
+	const saveLabel = ( product && ! isObject( product.id )
+		? translate( 'Update' )
+		: translate( 'Save & Publish' )
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-header.js]: Hi! I've found a possible matching string that has already been translated **67** times:
`translate( 'Save &amp; Publish' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/products/product-update.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+	onTrash = () => {
+		// TODO: Add action dispatch to trash this product.
+	}
+
+	onSave = () => {
+		const { product, translate } = this.props;
+
+		const successAction = successNotice(
+			translate( '%(product)s successfully updated.', {
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-update.js]: Hi! I've found a possible matching string that has already been translated **57** times:
`translate( '%1$s updated successfully.' )` *ES Score: 10*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/product-action-list-updates)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Pull request merged**